cpdb/cpdb-frontend.c

```diff
@@ -492,28 +505,105 @@ cpdb_printer_obj_t *cpdbRemovePrinter(cpdb_frontend_obj_t *f,
     return p;
 }
 
+void cpdbPrintBasicOptions(const cpdb_printer_obj_t *p)
+{
+    printf("-------------------------\n");
+    printf("Printer %s\n", p->id);
+    printf("name: %s\n", p->name);
+    printf("location: %s\n", p->location);
+    printf("info: %s\n", p->info);
+    printf("make and model: %s\n", p->make_and_model);
+    printf("accepting jobs? %s\n", (p->accepting_jobs ? "yes" : "no"));
+    printf("state: %s\n", p->state);
+    printf("backend: %s\n", p->backend_name);
+    printf("-------------------------\n\n");
+}
+
+void getAllPrintersLookup(gpointer key, gpointer value, gpointer user_data){
+    PrintBackend *proxy = value;
+    GError *error = NULL; 
+    
+    int num_printers;
+    GVariantIter iter;
+    GVariant *printers, *printer;
+    cpdb_printer_obj_t *p;
+
+    print_backend_call_get_filtered_printer_list_sync (proxy, &num_printers,
+                                                &printers, NULL, &error);
+    if (error)
+    {
+        logerror("Error getting printer list : %s\n", error->message);
+        return;
+    }
+    g_variant_iter_init(&iter, printers);
+    while (g_variant_iter_loop(&iter, "(v)", &printer))
+    {
+        p = cpdbGetNewPrinterObj();
+        cpdbFillBasicOptions(p, printer);
+        cpdbPrintBasicOptions(p);
+    }
+}
+
+void cpdbGetAllPrinters(cpdb_frontend_obj_t *f)
+{
+    loginfo("Hiding remote printers\n");
```

Comment:
According to its name, this function is to just list all the printers available, using whatever filter settings which are currently used, not for hiding the remote printers or list with remote printers hidden. So you should remove or modify this log line (probably copy-and-paste error).